design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+
+        - name: MY_POD_NAMESPACE
+          valueFrom:
+          fieldRef:
+            apiVersion: v1
+            fieldPath: metadata.namespace
+
+        - name: CONJUR_APPLIANCE_URL
+          value:  {{ .Values.secrets-provider-environment.conjurApplianceUrl }}
+
+        - name: CONJUR_AUTHN_URL
+          value:  {{ .Values.secrets-provider-environment.conjurAuthnUrl }}
+
+        - name: CONJUR_ACCOUNT
+          value:  {{ .Values.secrets-provider-environment.serviceAccount }}
+
+        - name: CONJUR_SSL_CERTIFICATE
+       	  valueFrom:
+            configMapKeyRef:
+        	  name: {{ .Values.secrets-provider-environment.conjurSslCertificate }}
+       		  key: ssl-certificate
+
+        - name: DEBUG
+       	  value:  {{ .Values.secrets-provider-environment.debug }}
+
+        - name: CONJUR_AUTHN_LOGIN
+       	  value:  {{ Values.secrets-provider-environment.conjurAuthnLogin }}
+
+        {{- if eq .Values.secrets-provider-environment.rotationSupport "true" }}
+        - name: SYNC_INTERVAL_TIME
+          value: {{ .Values.secrets-provider-environment.syncInternalTime }}
+        {{- end }}
+
+        {{- if .Values.secrets-provider-environment.k8SecretsLabels }}
+        - name: K8S_SECRETS_LABELS
+          value: {{ .Values.secrets-provider-environment.k8sSecretsLabels }}
+        {{- end }}
+```
+
+Note that when we transition to Milestone 2 we will need to keep the following in mind:
+
+`rotationSupport=false` default will need to be updated to`rotationSupport=true`. 
+
+We will need to add `syncIntervalTime=300` (5 minutes) to the default `values.yaml` where `syncIntervalTime` will represent how often the Secrets Provider will attempt to fetch new Conjur secrets. If the customer would like to change this parameter, they will need to do so manually by *reinstalling* *the Secrets Provider Chart*. 
+
+`ttlSecondsAfterFinished` will need to be removed from defaults
+
+All these changes *demands* that the customer manually deletes and installs the latest Helm Charts
+
+#### Packaging Helm Charts
+
+Every Helm project requires a `Chart.yml` file that describes the project. The Secrets Provider `Chart.yaml` wil resemble the following:
+
+```yaml
+apiVersion: v1
+description: A Helm chart for CyberArk Secrets Provider for Kubernetes
+home: https://www.cyberark.com
+icon: https://xebialabs-clients-iglusjax.stackpathdns.com/assets/files/logos/CyberArkConjurLogoWhiteBlue.png
+keywords:
+- security
+- secrets management
+maintainers:
+- email: conj_maintainers@cyberark.com
+  name: Conjur Maintainers
+name: secrets-provider
+version: 1.0.0
+```
+
+Helm Charts are packaged in `tgz` format. 
+
+For each release, we will package our Chart (`helm package secrets-provider`) and the output will be `secrets-provider-<version>.tgz`. This `.tgz` is what we will push to Github.
+
+For more detailed information on how a customer will install our Chart, see the section on [Installation](#installation).
+
+### Milestone 2: Rotation
```

Comment:
should we review this?

Reply:
Nothing to review yet for this one. There are Milestone 2 crumbs sprinkled across the doc but will hope to finalize this section today

Reply:
@orenbm Now you can...